biodatasets/dataset.py

```diff
@@ -0,0 +1,96 @@
+"""
+This module aims to define the Dataset class and the load_dataset api function.
+"""
+from typing import List
+from typing import Optional
+from typing import Tuple
+
+import numpy as np
+import pandas as pd
+
+from biodatasets.utils import logger
+from biodatasets.utils.google_bucket import get_dataset_path
+from biodatasets.utils.google_bucket import list_blobs
+from biodatasets.utils.google_bucket import pull_dataset
+
+log = logger.get(__name__)
+
+
+class Dataset:
+    """
+    Dataset class.
+    """
+
+    def __init__(self, name: str, force: bool = False):
+        """
+        Init the Dataset instance by fetching the dataset files.
+
+        :param name: name of the dataset
+        :param force: force fetching the dataset
+        """
+        self.name = name
+        self.path = get_dataset_path(name)
+        if not self.path.exists():
+            pull_dataset(name, force=force)
+
+    def to_npy_arrays(
+        self, input_names: List, target_names: Optional[List] = None
```

Comment:
To be complete you can add the expected type of each item in the list:
```suggestion
        self, input_names: List[str], target_names: Optional[List[str]] = None
```